### PR TITLE
Fix invalid enum references in groupings (fixes #327)

### LINKED ISSUE #327: Invalid enums referenced by various groups.
I would PR this straight in, but I need to confirm whether these are incorrect references or missing references.

cc:
NV @pdaniell-nv 
ARB @oddhack 
Silk.NET @frederikja163 

```
GL_RGB_NV
GL_ALPHA_NV
GL_FLOAT_NV
GL_DOUBLE_NV
GL_PROGRAM_STRING
GL_UNSIGNED_BYTE_SUN
and more...
```

Writing this, it seems that it's been assumed that all enums within groups with vendor suffixes too have suffixes. What's the best way to resolve this? Remove the suffixes from the enum group names, or just fix the references within the groups?

### PR DESCRIPTION
**Pull request by Perksey**
Fixes #327

### CODE REVIEW

File: xml/gl.xml [hunk 1]
```diff
@@ -134,11 +134,12 @@ typedef unsigned int GLhandleARB;
         </group>
 
         <group name="ProgramStringProperty">
-            <enum name="GL_PROGRAM_STRING"/>
+            <!-- Note to reviewer: could be PROGRAM_STRING_NV -->
```
Comment (Perksey) [inline xml/gl.xml]: Let me know what you think about this. I'll remove this comment once approved, but I wasn't sure whether to use GL_PROGRAM_STRING_ARB or GL_PROGRAM_STRING_NV (or both?), so I opted for ARB instead.

```suggestion
```

**Comment by pdaniell-nv** [comment]
Approved to merge.

File: xml/gl.xml (hunk 1 above)
Comment (pdaniell-nv) [reply xml/gl.xml]: That's weird. I thought I made a comment, but maybe I hit the wrong button. I would go for _ARB.

Comment (Perksey) [reply xml/gl.xml]: Yeah I thought that'd be the best thing to do. Thanks +1

**Comment by pdaniell-nv** [comment]
@oddhack this can be merged if you're happy with it.

File: xml/gl.xml [hunk 2]
```diff
@@ -176,44 +177,44 @@ typedef unsigned int GLhandleARB;
...
-            <enum name="GL_AMBIENT_SGIX"/>
-            <enum name="GL_DIFFUSE_SGIX"/>
-            <enum name="GL_SPECULAR_SGIX"/>
-            <enum name="GL_POSITION_SGIX"/>
-            <enum name="GL_SPOT_DIRECTION_SGIX"/>
-            <enum name="GL_SPOT_EXPONENT_SGIX"/>
-            <enum name="GL_SPOT_CUTOFF_SGIX"/>
-            <enum name="GL_CONSTANT_ATTENUATION_SGIX"/>
-            <enum name="GL_LINEAR_ATTENUATION_SGIX"/>
-            <enum name="GL_QUADRATIC_ATTENUATION_SGIX"/>
```
[the remark this answers is not in the document]
Comment (Perksey) [reply xml/gl.xml]: Fixed

File: xml/gl.xml [hunk 3]
```diff
@@ -269,32 +270,32 @@ typedef unsigned int GLhandleARB;
...
-            <enum name="GL_FLOAT_EXT"/>
-            <enum name="GL_DOUBLE_EXT"/>
+            <enum name="GL_BYTE"/>
+            <enum name="GL_SHORT"/>
+            <enum name="GL_FLOAT"/>
+            <enum name="GL_DOUBLE"/>
         </group>
 
         <group name="BinormalPointerTypeEXT">
-            <enum name="GL_BYTE_EXT"/>
```
[the remark this answers is not in the document]
Comment (Perksey) [reply xml/gl.xml]: Fixed.

**Comment by Perksey** [comment]
All comments have been addressed.

**Pull request merged**